tests/contrib/test_kubernetes/test_base.py

```diff
@@ -0,0 +1,124 @@
+import os
+import re
+import subprocess
+import tempfile
+import time
+
+import numpy as np
+import pytest
+from kubernetes import config
+from sklearn.datasets import load_iris
+
+from mlem.config import project_config
+from mlem.contrib.docker.base import DockerDaemon, DockerEnv
+from mlem.contrib.docker.helpers import build_model_image
+from mlem.contrib.kubernetes.base import (
+    K8sDeployment,
+    K8sDeploymentState,
+    K8sEnv,
+)
+from mlem.core.objects import DeployStatus
+from tests.contrib.test_kubernetes.conftest import k8s_test
+from tests.contrib.test_kubernetes.utils import Command
+
+
+@pytest.fixture
+def minikube_env_variables():
+    output = subprocess.check_output(
+        ["minikube", "-p", "minikube", "docker-env"]
+    )
+    export_re = re.compile('export ([A-Z_]+)="(.*)"\\n')
+    export_pairs = export_re.findall(output.decode("UTF-8"))
+    for k, v in export_pairs:
+        os.environ[k] = v
+
+
+@pytest.fixture
+def load_kube_config():
+    config.load_kube_config(os.getenv("KUBECONFIG", default="~/.kube/config"))
+
+
+@pytest.fixture
+def k8s_deployment(minikube_env_variables, model_meta_saved_single):
+    return K8sDeployment(
+        name="ml",
+        model=model_meta_saved_single.make_link(),
+        image_pull_policy="Never",
+        service_type="LoadBalancer",
```

Comment:
need to test other value too

Reply:
hmm, the mechanism of how EKS works and minikube works is different. Even if I determine host and port correctly in minikube for `nodeport`, one still cannot access it because it's inside a VM.

so we need to use a separate command that spits out a URL
and since we cannot get the relevant client, the only way to test it would be using something like `requests` (i.e. equivalent to `curl`)

one of the reasons this is complicated is because `minikube` runs only one node i.e. the control plane and not the worker nodes.

so given all the complexity, I would suggest to skip this.